src/Form/Type/PreAddAssetType.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Form\Type;
+
+use App\Entity\AssetType;
+use App\Entity\Organization;
+use App\Repository\AssetTypeRepository;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class PreAddAssetType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $parentOrganization = $options['parent_organization'];
+        if (!$parentOrganization instanceof Organization) {
+            throw new \LogicException('parent_organization must be an instance of Organization');
```

Comment:
this could have been done with `OptionResolver`: `$resolver->addAllowedTypes('parent_organization', Organization::class);`